extensions-contrib/virtual-columns/src/main/java/org/apache/druid/segment/MapVirtualColumnDimensionSelector.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.druid.segment;
+
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+
+abstract class MapVirtualColumnDimensionSelector implements DimensionSelector
+{
+  private final DimensionSelector keySelector;
+  private final DimensionSelector valueSelector;
+
+  MapVirtualColumnDimensionSelector(
+      DimensionSelector keySelector,
+      DimensionSelector valueSelector
+  )
+  {
+    this.keySelector = keySelector;
+    this.valueSelector = valueSelector;
+  }
+
+  protected DimensionSelector getKeySelector()
+  {
+    return keySelector;
+  }
+
+  protected DimensionSelector getValueSelector()
+  {
+    return valueSelector;
+  }
+
+  @Override
+  public double getDouble()
+  {
+    assert NullHandling.replaceWithDefault();
+    return 0.0;
```

Comment:
nit: probably better to throw unsupported exceptions here or atleast have an assertion that always fails, but since it had the same behavior prior to this PR may be we can make this change in a subsequent PR.

Reply:
Sounds good. I think it would be a more general discussion about how to support `Map` type in Druid.